src/python/pants/option/optionable.py

```diff
@@ -28,6 +45,130 @@ class Optionable(AbstractClass):
   deprecated_options_scope = None
   deprecated_options_scope_removal_version = None
 
+  @classmethod
+  def implementation_versions(cls):
```

Comment:
This is a good change... please update https://github.com/pantsbuild/pants/blob/d5237361e480dba1a2d1a0b4865bc7a235bc535b/src/docs/dev_tasks.md#task-implementation-versioning to account for it.

Reply:
I've [updated `dev_tasks.md`](https://github.com/pantsbuild/pants/pull/5161/files#diff-3e103fe1b1972f0a779f610cf524f7fb) -- let me know if this is the information we'd want there.